### PR TITLE
misc improvements (caching, logging, robustness)

### PR DESCRIPTION
**Pull request by turadg**
Ancillary changes for https://github.com/Agoric/agoric-3-proposals/pull/88 that needn't wait for that PR itself to be landed.

### CODE REVIEW

File: packages/synthetic-chain/upgrade-test-scripts/run_test.sh [hunk 1]
```diff
@@ -6,12 +6,20 @@ set -e
...
-PROPOSAL_PATH=$1
+PROPOSAL=$1
+if [ -z "$PROPOSAL" ]; then
+    echo "Must specify what proposal to use"
+    exit 1
+fi
+
+echo "[$PROPOSAL] Starting agd"
+
+echo "[$PROPOSAL_PATH] Starting agd"
```
Comment (michaelfig) [inline packages/synthetic-chain/upgrade-test-scripts/run_test.sh:17]: Detritus?

```suggestion
```

File: packages/synthetic-chain/upgrade-test-scripts/run_use.sh [hunk 2]
```diff
@@ -4,31 +4,31 @@
...
 
-ID=$1
-if [ -z "$ID" ]; then
+PROPOSAL=$1
+if [ -z "$PROPOSAL" ]; then
     echo "Must specify what proposal to use"
     exit 1
 fi
-PROPOSAL_PATH="/usr/src/proposals/$ID/"
+PROPOSAL_PROPOSAL="/usr/src/proposals/$PROPOSAL/"
```
Comment (michaelfig) [inline packages/synthetic-chain/upgrade-test-scripts/run_use.sh:12]: Maybe:
```suggestion
PROPOSAL_DIR="/usr/src/proposals/$PROPOSAL/"
```

**Review by michaelfig — APPROVED** [review]
LGTM!  A few minor suggestions, but the logic looks sound.

Thanks for separating this into a more digestible form.

File: packages/synthetic-chain/upgrade-test-scripts/run_test.sh (hunk 1 above)
Comment (turadg) [reply packages/synthetic-chain/upgrade-test-scripts/run_test.sh:17]: ah, probably bad rebase. will correct in next PR

File: packages/synthetic-chain/src/lib/agd-lib.ts
[the remark this answers is not in the document]
Comment (turadg) [reply packages/synthetic-chain/src/lib/agd-lib.ts:1]: https://github.com/Agoric/agoric-3-proposals/pull/102

**Pull request merged**